src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -0,0 +1,1218 @@
+            }
+
+            final int codingStartPos = FuncotatorUtils.getStartPositionInTranscript(variant, activeRegions, strand);
+
+            //Check for de novo starts:
+            if ( FuncotatorUtils.getEukaryoticAminoAcidByCodon(referenceCodingSequence.substring(codingStartPos, codingStartPos+3) )
+                    == AminoAcid.METHIONINE ) {
+
+                // We know we have a new start.
+                // Is it in frame or out of frame?
+                if ( FuncotatorUtils.isInFrameWithEndOfRegion(codingStartPos, referenceCodingSequence.length()) ) {
+                    gencodeFuncotation.setVariantClassification(GencodeFuncotation.VariantClassification.DE_NOVO_START_IN_FRAME);
+                }
+                else {
+                    gencodeFuncotation.setVariantClassification(GencodeFuncotation.VariantClassification.DE_NOVO_START_OUT_FRAME);
+                }
+            }
+            else {
+                gencodeFuncotation.setVariantClassification(GencodeFuncotation.VariantClassification.FIVE_PRIME_UTR);
+            }
+        }
+        else {
+            gencodeFuncotation.setVariantClassification(GencodeFuncotation.VariantClassification.THREE_PRIME_UTR);
+        }
+
+        return gencodeFuncotation;
+    }
+
+    /**
+     * Create a {@link GencodeFuncotation} for a {@code variant} that occurs in an intron in the given {@code transcript}.
+     * @param variant The {@link VariantContext} for which to create a {@link GencodeFuncotation}.
+     * @param altAllele The {@link Allele} in the given {@code variant} for which to create a {@link GencodeFuncotation}.
+     * @param gtfFeature The {@link GencodeGtfGeneFeature} in which the given {@code variant} occurs.
+     * @param transcript The {@link GencodeGtfTranscriptFeature} in which the given {@code variant} occurs.
+     * @return A {@link GencodeFuncotation} containing information about the given {@code variant} given the corresponding {@code transcript}.
+     */
+    private static GencodeFuncotation createIntronFuncotation(final VariantContext variant,
+                                                       final Allele altAllele,
+                                                       final GencodeGtfGeneFeature gtfFeature,
+                                                       final GencodeGtfTranscriptFeature transcript) {
+
+        // Setup the "trivial" fields of the gencodeFuncotation:
+        final GencodeFuncotation gencodeFuncotation = createGencodeFuncotationWithTrivialFieldsPopulated(variant, altAllele, gtfFeature, transcript);
+
+        // Set as default INTRON variant classification:
+        gencodeFuncotation.setVariantClassification(GencodeFuncotation.VariantClassification.INTRON);
+
+        // Determine the strand for the variant:
+        final Strand strand = Strand.toStrand( transcript.getGenomicStrand().toString() );
+        if ( strand == Strand.NONE ) {
```

Comment:
Can you use a more specific GATKException?

Reply:
Sure.  I made it a `UserException.MalformedFile` because the strand should come from the transcript in the file itself.